cmd/newtemplates/templates.go

```diff
@@ -0,0 +1,74 @@
+// Package templates provides the templates for the k6 new command
+package templates
+
+import (
+	_ "embed"
+	"fmt"
+	"io"
+	"text/template"
+)
+
+//go:embed minimal.js
+var minimalTemplateContent string
+
+//go:embed protocol.js
+var protocolTemplateContent string
+
+//go:embed browser.js
+var browserTemplateContent string
+
+// TemplateManager manages the pre-parsed templates
+type TemplateManager struct {
+	minimalTemplate  *template.Template
+	protocolTemplate *template.Template
+	browserTemplate  *template.Template
+}
+
+// NewTemplateManager initializes a new TemplateManager with parsed templates
+func NewTemplateManager() (*TemplateManager, error) {
+	minimalTmpl, err := template.New("minimal").Parse(minimalTemplateContent)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse minimal template: %w", err)
+	}
+
+	protocolTmpl, err := template.New("protocol").Parse(protocolTemplateContent)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse protocol template: %w", err)
+	}
+
+	browserTmpl, err := template.New("browser").Parse(browserTemplateContent)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse browser template: %w", err)
+	}
+
+	return &TemplateManager{
+		minimalTemplate:  minimalTmpl,
+		protocolTemplate: protocolTmpl,
+		browserTemplate:  browserTmpl,
+	}, nil
+}
+
+// GetTemplate selects the appropriate template based on the type
+func (tm *TemplateManager) GetTemplate(templateType string) (*template.Template, error) {
+	switch templateType {
```

Comment:
Nit; I guess we could define the available "types" as exposed constants from the package, so it's a bit more explicit. But feel free to disregard this comment, cause for now it's only used by `new`, and very concretely.

Reply:
I'd pretty much like that too 👍🏻 Having them as constants, would help defining them as keys we explicitly own (and also help with later refactorings).

Reply:
Makes sense. Done!